transport/whisper/types/received_message.go

```diff
@@ -0,0 +1,30 @@
+package whispertypes
+
+import (
+	"crypto/ecdsa"
+)
+
+// ReceivedMessage represents a data packet to be received through the
+// Whisper protocol and successfully decrypted.
+type ReceivedMessage interface {
+	Raw() []byte
```

Comment:
Is `Raw()` here the full raw message? Why is that needed?
Is `Signature()` needed? After all there is already `Src` which gets recovered from the signature.
Which salt is `Salt()` returning?

Furthermore, in the Nimbus Whisper implementation we do not expose `P2P` and `SymKeyHash`. Are both needed?
Important side note that I haven't mentioned before: We do not have a mailserver implementation at the moment.

Reply:
See explanation above. Will be removing unused functions next.

Reply:
I see a lot got removed, but OK if that is all what is required.